app/src/test/java/com/codesoom/assignment/controllers/UserControllerTest.java

```diff
@@ -0,0 +1,152 @@
+@DisplayName("UserController 테스트")
+class UserControllerTest {
+
+    private MockMvc mvc;
+    @Autowired
+    private UserController controller;
+    @Autowired
+    private UserService service;
+    @Autowired
+    private ObjectMapper mapper;
+
+    @BeforeEach
+    void setUp() {
+        mvc = MockMvcBuilders
+                .standaloneSetup(controller)
+                .setControllerAdvice(new NotFoundErrorAdvice())
+                .build();
+    }
+
+    @Nested
+    @DisplayName("findAll 메서드는")
+    class Describe_FindAll{
+
+        @Nested
+        @DisplayName("사용자가 존재 한다면")
+        class Context_ExistedUsers{
+
+            private List<User> users;
+
+            @BeforeEach
+            void setUp() {
+                users = addNewUsers(service , TEST_SIZE);
+            }
+
+            @AfterEach
+            void tearDown() {
+                repositoryClear(service);
+            }
+
+            @Test
+            @DisplayName("모든 사용자를 반환한다")
+            void It_ReturnUsers() throws Exception {
+                String content = mapper.writeValueAsString(users);
+                System.out.println(content);
+                mvc.perform(get("/user"))
+                        .andExpect(status().isOk())
+                        .andExpect(content().string(equalTo(content)));
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("create 메서드는")
+    class Describe_Create{
+
+        @Nested
+        @DisplayName("필수 입력 항목이 다 존재한다면")
+        class Context_PassValidation{
+
+            private User user;
+
+            @BeforeEach
+            void setUp() {
+            }
+
+            @Test
+            @DisplayName("저장 후 저장한 정보를 반환한다")
+            void It_Save(){
+
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("update 메서드는")
+    class Describe_Update{
+
+        @Nested
+        @DisplayName("{id}에 해당하는 자원이 있고 , 수정할 정보가 비어있지 않다면")
+        class Context_ExistedIdAndNotEmptyBody{
+
+            private final Long id = 1L;
+            private User updateUser;
+
+            @BeforeEach
+            void setUp() {
+                updateUser = User.builder()
+                                .id(id)
+                                .name("UPDATE")
+                                .email("UPDATE")
+                                .password("UPDATE")
+                                .build();
+            }
+
+            @Test
+            @DisplayName("수정 후 수정된 정보를 반환한다")
+            void It_UpdateUser(){
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("delete 메서드는")
+    class Describe_Delete{
+
+        @Nested
+        @DisplayName("{id}에 해당하는 자원이 존재한다면")
+        class Context_ExistedId{
+
+            private final Long id = 1L;
+            private User user;
+
+            @BeforeEach
+            void setUp() {
+            }
+
+            @Test
+            @DisplayName("삭제 후 삭제한 정보를 반환한다")
+            void It_Delete(){
+            }
+        }
+    }
```

Comment:
테스트 코드를 잊지 않고 만들기 위해 이렇게 준비를 해두셨군요. 좋습니다.
보통 이렇게 미리 테스트를 만들어 두는 경우에는 잊거나 빠뜨리지 않게 일부러 테스트를 실패하게 만들곤 합니다.
```suggestion
    @Nested
    @DisplayName("delete 메서드는")
    class Describe_Delete{

        @Nested
        @DisplayName("{id}에 해당하는 자원이 존재한다면")
        class Context_ExistedId{

            private final Long id = 1L;
            private User user;

            @BeforeEach
            void setUp() {
            }

            @Test
            @DisplayName("삭제 후 삭제한 정보를 반환한다")
            void It_Delete(){
                fail("미구현된 테스트입니다.");
            }
        }
    }
```